ktor-server/ktor-server-core/jvm/src/io/ktor/routing/RouteSelector.kt

```diff
@@ -28,6 +28,11 @@ data class RouteSelectorEvaluation(val succeeded: Boolean,
          */
         const val qualityConstant = 1.0
 
+        /**
+         * Quality of [RouteSelectorEvaluation] when a query parameter has matched
+         */
+        internal const val qualityQueryParameter = 1.0
```

Comment:
Should we order selection between `qualityConstant` and `qualityQueryParameter`?

Could you write the test case with both on the same level?